astroquery/cadc/core.py

```diff
@@ -137,20 +132,65 @@ def login(self, user=None, password=None, certificate_file=None):
             password to login with
         certificate : str, required if user is None
             path to certificate to use with logging in
+
+        Notes
+        -----
+        This will soon be deprecated as it does not make sense to login with
+        certificates.
         """
-        login_url = get_access_url(self.CADCLOGIN_SERVICE_URI,
-                                   'ivo://ivoa.net/std/UMS#login-0.1')
-        return self._cadctap.login(user=user, password=password,
-                                   certificate_file=certificate_file,
-                                   cookie_prefix=CADC_COOKIE_PREFIX,
-                                   login_url=login_url,
-                                   verbose=False)
-
-    def logout(self, verbose=False):
+        if certificate_file:
+            # In order to force an HTTPS hand shake on the session is to
+            # recreate it.
+            pyvo.dal.tap.s = requests.Session()
+            pyvo.dal.tap.s.cert = certificate_file
+        elif not user or not password:
+            raise AttributeError('login credentials missing (user/password '
+                                 'or certificate)')
+        else:
+            login_url = get_access_url(self.CADCLOGIN_SERVICE_URI,
+                                       'ivo://ivoa.net/std/UMS#login-0.1')
+            if login_url is None:
+                raise RuntimeError("No login URL")
+            # need to login and get a cookie
+            args = {
+                "username": str(user),
+                "password": str(password)}
+            header = {
+                "Content-type": "application/x-www-form-urlencoded",
+                "Accept": "text/plain"
+            }
+            response = requests.post(login_url, data=args, headers=header)
```

Comment:
any reason this isn't using the CADC object's own session, or the pyvo session defined above?